apps/system/js/bluetooth_v2.js

```diff
@@ -0,0 +1,438 @@
+     *   1. connecting with a headset
+     *   2. transfering a file to/from another device
+     * So we need to listen to corresponding events to know we are (aren't)
+     * connected, then summarize to an event and dispatch to StatusBar
+     */
+    // In headset connected case:
+    adapter.addEventListener('hfpstatuschanged', (evt) => {
+      this._setProfileConnected(this.Profiles.HFP, evt.status);
+    });
+
+    adapter.addEventListener('a2dpstatuschanged', (evt) => {
+      this._setProfileConnected(this.Profiles.A2DP, evt.status);
+    });
+
+    adapter.addEventListener('scostatuschanged', (evt) => {
+      this._setProfileConnected(this.Profiles.SCO, evt.status);
+    });
+  },
+
+  /**
+   * BT APIv2: Watch 'onattributechanged' event from
+   * mozBluetooth.defaultAdapter for updating state information.
+   *
+   * @private
+   * @param  {Object} evt event object
+   */
+  _bluetoothAdapterHandler:
+    function bt__bluetoothAdapterHandler(evt) {
+      for (var i in evt.attrs) {
+        switch (evt.attrs[i]) {
+          case 'state':
+            if (this._adapter.state === 'enabled') {
+              this._isEnabled = true;
+              navigator.mozSettings.createLock()
+                .set({'bluetooth.enabled': true});
+              window.dispatchEvent(new CustomEvent('bluetooth-enabled'));
+              // this.icon && this.icon.update();
+            } else if (this._adapter.state === 'disabled') {
+              this._isEnabled = false;
+              navigator.mozSettings.createLock()
+                .set({'bluetooth.enabled': false});
+              window.dispatchEvent(new CustomEvent('bluetooth-disabled'));
+              // this.icon && this.icon.update();
+            }
+            break;
+          default:
+            break;
+        }
+      }
+  },
+
+  /**
+   * Watch 'onattributechanged' event from mozBluetooth for updating default
+   * adapter information.
+   *
+   * 'onattributechanged' event description:
+   * A handler to trigger when bluetooth manager's only property
+   * defaultAdapter has changed.
+   *
+   * @private
+   * @param  {Object} evt event object
+   */
+  _bluetoothManagerHandler:
+    function bt__bluetoothManagerHandler(resolve, evt) {
+      for (var i in evt.attrs) {
+        switch (evt.attrs[i]) {
+          case 'defaultAdapter':
+            this._debug('defaultAdapter changed.');
+            if (this._bluetooth.defaultAdapter) {
+              // Default adapter attribute change.
+              // Usually, it means that we reach new default adapter.
+              this._adapter = this._bluetooth.defaultAdapter;
+              this._isEnabled = (this._adapter.state === 'enabled');
+              this._stateChangeHandler =
+                this._bluetoothAdapterHandler.bind(this);
+              this._adapter.addEventListener('attributechanged',
+                this._stateChangeHandler);
+              this._updateProfileState(this._adapter);
```

Comment:
Per the implementation of  `_updateProfileState`, it add event listener here. But there is no remove event listener while the default adapter is removed.